core/unit_test/TestKokkosHelpCausesNormalProgramTermination.cpp

```diff
@@ -0,0 +1,50 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+#include "KokkosExecutionEnvironmentNeverInitializedFixture.hpp"
+
+namespace {
+
+using KokkosHelpCausesNormalProgramTermination_DeathTest =
+    KokkosExecutionEnvironmentNeverInitialized;
+
+TEST_F(KokkosHelpCausesNormalProgramTermination_DeathTest,
+       print_help_and_exit_early) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  int argc = 1;
+
+  char const *argv[] = {
+      "--kokkos-help",
```

Comment:
```suggestion
      "--kokkos-help", nullptr,
```
`argv` is assumed to be nullptr terminated, if not there's a stack buffer overflow (reported by asan) when removing the help flag from `argv`
https://github.com/kokkos/kokkos/blob/152560be94635163cc9ed5bbfd74a06fafb68bc3/core/src/impl/Kokkos_Core.cpp#L911-L916)

Reply:
Yup.  Good catch.  I have already pushed the fix.